### ISSUE TITLE
add_menu_page expecting a callable but callable|callable-string should be fine too

### ISSUE THREAD

**Issue by herndlm** [szepeviktor/phpstan-wordpress#85]
Is this a problem we can fix? Looks like it's coming from the stubs / wordpress docs, but actually it is supposed to be used with callable strings, right? E.g. as in 
```php
add_menu_page('Docs', 'Docs', 'manage_options', 'docs', 'my-docs-function', 'dashicons-editor-help');
```

phpstan reports
```
Parameter #5 $function add_menu_page expects callable(): mixed, 'my-docs-function' given.
```

I guess many many more functions are having the same problems?

This is more like a discussion, @szepeviktor want to enable the discussions feature in this repo? ;)

**Comment by szepeviktor** [comment]
Let's keep everything in issues.

:) Is `callable-string` a Martin-invention?
PHPStan is a tool for OOP. Global function names are not supported as callable yet. There is an open issues for this.

**Comment by szepeviktor** [comment]
In normal OOP world `'my-docs-function'` is `[MyClass::class, 'myDocsFunction']` or `[$this, 'myDocsFunction']`.

**Comment by johnbillion** [comment]
`my-docs-function` isn't a valid function name so it's not a valid callable. Should it be `my_docs_function` ?

**Comment by herndlm** [comment]
`callable-string` is real :) 

And yeah this was an invalid example, sorry. Was reported by a colleague and the default value of '' is also is not accepted. Not a 100% sure if it would accept valid function names, but I think it doesn't

I prefer real callables but I think that's not the WP way, right?

**Comment by johnbillion** [comment]
If you pass a valid callback function name to that parameter there should not be an error. What's the actual value you're passing? Is it the string name of a function that exists? `my_function` is a valid callable as long as that a function with that name exists. OOP has nothing to do with it.

**Comment by herndlm** [comment]
OK, thx. It's only about the default value of '' then. Which cannot be specified in case one of the later args needs to be set.

**Comment by johnbillion** [comment]
Ah yes the default value of `''` is a known issue, it should be nullable instead. I think that's covered in one of the PHPStan tickets for WordPress core, maybe https://core.trac.wordpress.org/ticket/52217

**Comment by herndlm** [comment]
Thx for the insights! That's enough for me :)
